### PR TITLE
Test application metric scraping

### PR DESCRIPTION
**Pull request by howardjohn**
* Fix the prom scraping config to check the right flag to enable secure
scraping
* Enable strict mTLS for the stats test -- permissive is broken
* Switch stats test to echo (from bookinfo)
* Add metric flags to the echo deployment

### CODE REVIEW

File: tests/integration/telemetry/stats/prometheus/http/stats.go [hunk 1]
```diff
@@ -93,61 +99,93 @@ func TestSetup(ctx resource.Context) (err error) {
...
 		return
 	}
-	err = galInst.ApplyConfig(
-		bookinfoNsInst,
-		bookingfoGatewayFile,
-	)
-	if err != nil {
-		return
+	// Application scraping will not work with permissive mode
+	if err = galInst.ApplyConfig(appNsInst, fmt.Sprintf(`
```
Comment (bianpengyuan) [inline tests/integration/telemetry/stats/prometheus/http/stats.go]: I think there is cluster wide option to turn on strict mtls? Could you instead add those options to these two places:
https://github.com/istio/istio/blob/b3b0fd35cae8dfd0321a7141f5313cc300b2523d/tests/integration/telemetry/stats/prometheus/http/wasm/stats_wasm_filter_test.go#L52 https://github.com/istio/istio/blob/b3b0fd35cae8dfd0321a7141f5313cc300b2523d/tests/integration/telemetry/stats/prometheus/http/nullvm/stats_filter_test.go#L51

File: tests/integration/telemetry/stats/prometheus/util_prometheus.go [hunk 2]
```diff
@@ -35,3 +38,20 @@ func QueryPrometheus(t *testing.T, query string, promInst prometheus.Instance) e
 	t.Logf("get value %v", got)
 	return nil
 }
+
+// QueryFirstPrometheus queries prometheus and returns the result once a timeseries exists
+func QueryFirstPrometheus(t *testing.T, query string, promInst prometheus.Instance) error {
+	t.Logf("query prometheus with: %v", query)
+	val, err := promInst.WaitForOneOrMore(query)
+	if err != nil {
+		log.Errorf("howardjohn: err1 %v", err)
```
Comment (bianpengyuan) [inline tests/integration/telemetry/stats/prometheus/util_prometheus.go]: Debug log?

File: tests/integration/telemetry/stats/prometheus/http/stats.go (hunk 1 above)
Comment (howardjohn) [reply tests/integration/telemetry/stats/prometheus/http/stats.go]: This option has been removed, this is now the "correct" way to apply mTLS

File: tests/integration/telemetry/stats/prometheus/util_prometheus.go (hunk 2 above)
Comment (howardjohn) [reply tests/integration/telemetry/stats/prometheus/util_prometheus.go]: Oops, thanks

**Review by bianpengyuan — APPROVED** [review]

**Comment by howardjohn** [comment]
This is to help test https://github.com/istio/istio/issues/21366 in the future

File: pkg/test/framework/components/echo/kube/deployment_test.go [hunk 3]
```diff
@@ -140,13 +140,13 @@ func TestDeploymentYAML(t *testing.T) {
 		gotBytes := []byte(serviceYAML + "---" + deploymentYAML)
 		wantedBytes := testutil.ReadGoldenFile(gotBytes, tc.wantFilePath, t)
 
-		wantBytes := testutil.StripVersion(wantedBytes)
-		gotBytes = testutil.StripVersion(gotBytes)
+			wantBytes := testutil.StripVersion(wantedBytes)
```
Comment (nmittler) [inline pkg/test/framework/components/echo/kube/deployment_test.go]: was this indentation intentional?

File: pkg/test/framework/components/prometheus/kube.go [hunk 4]
```diff
@@ -181,28 +181,34 @@ func (c *kubeComponent) WaitForOneOrMore(format string, args ...interface{}) err
...
+			return v, true, nil
 
 		default:
 			return nil, true, fmt.Errorf("unhandled value type: %v", v.Type())
 		}
 	}, retryTimeout, retryDelay)
 
-	return err
+	var v model.Value
+	if value != nil {
```
Comment (nmittler) [inline pkg/test/framework/components/prometheus/kube.go]: is the assumption here that if err == nil, value will always be a valid `model.Value`?

**Review by nmittler — APPROVED** [review]

File: pkg/test/framework/components/echo/kube/deployment_test.go (hunk 3 above)
Comment (howardjohn) [reply pkg/test/framework/components/echo/kube/deployment_test.go]: Too many rebases.. thanks

**Pull request merged**